MiniSim/Menu.swift

```diff
@@ -34,19 +34,16 @@ class Menu: NSMenu {
     }
     
     func getDevices() {
-        DispatchQueue.global(qos: .userInitiated).async {
-            do {
-                var devicesArray: [Device] = []
-                if UserDefaults.standard.enableiOSSimulators {
-                    try devicesArray.append(contentsOf: DeviceService.getIOSDevices())
-                }
-                if UserDefaults.standard.enableAndroidEmulators && UserDefaults.standard.androidHome != nil {
-                    try devicesArray.append(contentsOf: DeviceService.getAndroidDevices())
-                }
-                self.devices = devicesArray
-            } catch {
+        let userDefaults = UserDefaults.standard
+        DeviceService.getAllDevices(
+            android: userDefaults.enableAndroidEmulators && userDefaults.androidHome != nil,
+            iOS: userDefaults.enableiOSSimulators
+        ) { devices, error in
+            if let error = error {
```

Comment:
You can do `if let error` without the `= error`

Reply:
Thanks for the hint! Done.